// @aws-sdk/client-rds-data/dist-cjs/index.js
const { awsEndpointFunctions, emitWarningIfUnsupportedVersion: emitWarningIfUnsupportedVersion$1, createDefaultUserAgentProvider, NODE_APP_ID_CONFIG_OPTIONS, getAwsRegionExtensionConfiguration, resolveAwsRegionExtensionConfiguration, resolveUserAgentConfig, resolveHostHeaderConfig, getUserAgentPlugin, getHostHeaderPlugin, getLoggerPlugin, getRecursionDetectionPlugin } = require("@aws-sdk/core/client");
const { getHttpAuthSchemeEndpointRuleSetPlugin, DefaultIdentityProviderConfig, getHttpSigningPlugin } = require("@smithy/core");
const { normalizeProvider, getSmithyContext, ServiceException, NoOpLogger, emitWarningIfUnsupportedVersion, loadConfigsForDefaultMode, getDefaultExtensionConfiguration, resolveDefaultRuntimeConfig, Client, makeBuilder, createAggregatedClient } = require("@smithy/core/client");
const { Command: $Command } = require("@smithy/core/client");
exports.$Command = $Command;
exports.__Client = Client;
const { resolveDefaultsModeConfig, loadConfig, NODE_USE_FIPS_ENDPOINT_CONFIG_OPTIONS, NODE_USE_DUALSTACK_ENDPOINT_CONFIG_OPTIONS, NODE_REGION_CONFIG_OPTIONS, NODE_REGION_CONFIG_FILE_OPTIONS, resolveRegionConfig } = require("@smithy/core/config");
const { BinaryDecisionDiagram, EndpointCache, decideEndpoint, customEndpointFunctions, resolveEndpointConfig, getEndpointPlugin } = require("@smithy/core/endpoints");
const { parseUrl, getHttpHandlerExtensionConfiguration, resolveHttpHandlerRuntimeConfig, getContentLengthPlugin } = require("@smithy/core/protocols");
const { DEFAULT_RETRY_MODE, NODE_RETRY_MODE_CONFIG_OPTIONS, NODE_MAX_ATTEMPT_CONFIG_OPTIONS, resolveRetryConfig, getRetryPlugin } = require("@smithy/core/retry");
const { TypeRegistry, getSchemaSerdePlugin } = require("@smithy/core/schema");
const { resolveAwsSdkSigV4Config, AwsSdkSigV4Signer, NODE_AUTH_SCHEME_PREFERENCE_OPTIONS } = require("@aws-sdk/core/httpAuthSchemes");
const { defaultProvider } = require("@aws-sdk/credential-provider-node");
const { toUtf8, fromUtf8, toBase64, fromBase64, calculateBodyLength } = require("@smithy/core/serde");
const { streamCollector, NodeHttpHandler } = require("@smithy/node-http-handler");
const { AwsRestJsonProtocol } = require("@aws-sdk/core/protocols");
const { Sha256 } = require("@smithy/core/checksum");

const defaultRDSDataHttpAuthSchemeParametersProvider = async (config, context, input) => {
    return {
        operation: getSmithyContext(context).operation,
        region: await normalizeProvider(config.region)() || (() => {
            throw new Error("expected `region` to be configured for `aws.auth#sigv4`");
        })(),
    };
};
function createAwsAuthSigv4HttpAuthOption(authParameters) {
    return {
        schemeId: "aws.auth#sigv4",
        signingProperties: {
            name: "rds-data",
            region: authParameters.region,
        },
        propertiesExtractor: (config, context) => ({
            signingProperties: {
                config,
                context,
            },
        }),
    };
}
const defaultRDSDataHttpAuthSchemeProvider = (authParameters) => {
    const options = [];
    switch (authParameters.operation) {
        default: {
            options.push(createAwsAuthSigv4HttpAuthOption(authParameters));
        }
    }
    return options;
};
const resolveHttpAuthSchemeConfig = (config) => {
    const config_0 = resolveAwsSdkSigV4Config(config);
    return Object.assign(config_0, {
        authSchemePreference: normalizeProvider(config.authSchemePreference ?? []),
    });
};

const resolveClientEndpointParameters = (options) => {
    return Object.assign(options, {
        useDualstackEndpoint: options.useDualstackEndpoint ?? false,
        useFipsEndpoint: options.useFipsEndpoint ?? false,
        defaultSigningName: "rds-data",
    });
};
const commonParams = {
    UseFIPS: { type: "builtInParams", name: "useFipsEndpoint" },
    Endpoint: { type: "builtInParams", name: "endpoint" },
    Region: { type: "builtInParams", name: "region" },
    UseDualStack: { type: "builtInParams", name: "useDualstackEndpoint" },
};

var version = "3.1077.0";
var packageInfo = {
	version: version};

const k = "ref";
const a = -1, b = true, c = "isSet", d = "PartitionResult", e = "booleanEquals", f = "getAttr", g = { [k]: "Endpoint" }, h = { [k]: d }, i = {}, j = [{ [k]: "Region" }];
const _data = {
    conditions: [
        [c, [g]],
        [c, j],
        ["aws.partition", j, d],
        [e, [{ [k]: "UseFIPS" }, b]],
        [e, [{ [k]: "UseDualStack" }, b]],
        [e, [{ fn: f, argv: [h, "supportsDualStack"] }, b]],
        [e, [{ fn: f, argv: [h, "supportsFIPS"] }, b]]
    ],
    results: [
        [a],
        [a, "Invalid Configuration: FIPS and custom endpoint are not supported"],
        [a, "Invalid Configuration: Dualstack and custom endpoint are not supported"],
        [g, i],
        ["https://rds-data-fips.{Region}.{PartitionResult#dualStackDnsSuffix}", i],
        [a, "FIPS and DualStack are enabled, but this partition does not support one or both"],
        ["https://rds-data-fips.{Region}.{PartitionResult#dnsSuffix}", i],
        [a, "FIPS is enabled but this partition does not support FIPS"],
        ["https://rds-data.{Region}.{PartitionResult#dualStackDnsSuffix}", i],
        [a, "DualStack is enabled but this partition does not support DualStack"],
        ["https://rds-data.{Region}.{PartitionResult#dnsSuffix}", i],
        [a, "Invalid Configuration: Missing Region"]
    ]
};
const root = 2;
const r = 100_000_000;
const nodes = new Int32Array([
    -1, 1, -1,
    0, 12, 3,
    1, 4, r + 11,
    2, 5, r + 11,
    3, 8, 6,
    4, 7, r + 10,
    5, r + 8, r + 9,
    4, 10, 9,
    6, r + 6, r + 7,
    5, 11, r + 5,
    6, r + 4, r + 5,
    3, r + 1, 13,
    4, r + 2, r + 3,
]);
const bdd = BinaryDecisionDiagram.from(nodes, root, _data.conditions, _data.results);

const cache = new EndpointCache({
    size: 50,
    params: ["Endpoint", "Region", "UseDualStack", "UseFIPS"],
});
const defaultEndpointResolver = (endpointParams, context = {}) => {
    return cache.get(endpointParams, () => decideEndpoint(bdd, {
        endpointParams: endpointParams,
        logger: context.logger,
    }));
};
customEndpointFunctions.aws = awsEndpointFunctions;

class RDSDataServiceException extends ServiceException {
    constructor(options) {
        super(options);
        Object.setPrototypeOf(this, RDSDataServiceException.prototype);
    }
}

class AccessDeniedException extends RDSDataServiceException {
    name = "AccessDeniedException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "AccessDeniedException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, AccessDeniedException.prototype);
    }
}
class BadRequestException extends RDSDataServiceException {
    name = "BadRequestException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "BadRequestException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, BadRequestException.prototype);
    }
}
class DatabaseErrorException extends RDSDataServiceException {
    name = "DatabaseErrorException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "DatabaseErrorException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, DatabaseErrorException.prototype);
    }
}
class DatabaseNotFoundException extends RDSDataServiceException {
    name = "DatabaseNotFoundException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "DatabaseNotFoundException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, DatabaseNotFoundException.prototype);
    }
}
class DatabaseResumingException extends RDSDataServiceException {
    name = "DatabaseResumingException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "DatabaseResumingException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, DatabaseResumingException.prototype);
    }
}
class DatabaseUnavailableException extends RDSDataServiceException {
    name = "DatabaseUnavailableException";
    $fault = "server";
    constructor(opts) {
        super({
            name: "DatabaseUnavailableException",
            $fault: "server",
            ...opts,
        });
        Object.setPrototypeOf(this, DatabaseUnavailableException.prototype);
    }
}
class ForbiddenException extends RDSDataServiceException {
    name = "ForbiddenException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "ForbiddenException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ForbiddenException.prototype);
    }
}
class HttpEndpointNotEnabledException extends RDSDataServiceException {
    name = "HttpEndpointNotEnabledException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "HttpEndpointNotEnabledException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, HttpEndpointNotEnabledException.prototype);
    }
}
class InternalServerErrorException extends RDSDataServiceException {
    name = "InternalServerErrorException";
    $fault = "server";
    constructor(opts) {
        super({
            name: "InternalServerErrorException",
            $fault: "server",
            ...opts,
        });
        Object.setPrototypeOf(this, InternalServerErrorException.prototype);
    }
}
class InvalidResourceStateException extends RDSDataServiceException {
    name = "InvalidResourceStateException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "InvalidResourceStateException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidResourceStateException.prototype);
    }
}
class InvalidSecretException extends RDSDataServiceException {
    name = "InvalidSecretException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "InvalidSecretException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidSecretException.prototype);
    }
}
class SecretsErrorException extends RDSDataServiceException {
    name = "SecretsErrorException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "SecretsErrorException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, SecretsErrorException.prototype);
    }
}
class ServiceUnavailableError extends RDSDataServiceException {
    name = "ServiceUnavailableError";
    $fault = "server";
    constructor(opts) {
        super({
            name: "ServiceUnavailableError",
            $fault: "server",
            ...opts,
        });
        Object.setPrototypeOf(this, ServiceUnavailableError.prototype);
    }
}
class StatementTimeoutException extends RDSDataServiceException {
    name = "StatementTimeoutException";
    $fault = "client";
    dbConnectionId;
    constructor(opts) {
        super({
            name: "StatementTimeoutException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, StatementTimeoutException.prototype);
        this.dbConnectionId = opts.dbConnectionId;
    }
}
class TransactionNotFoundException extends RDSDataServiceException {
    name = "TransactionNotFoundException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "TransactionNotFoundException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, TransactionNotFoundException.prototype);
    }
}
class NotFoundException extends RDSDataServiceException {
    name = "NotFoundException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "NotFoundException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, NotFoundException.prototype);
    }
}
class UnsupportedResultException extends RDSDataServiceException {
    name = "UnsupportedResultException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "UnsupportedResultException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, UnsupportedResultException.prototype);
    }
}

const _ADE = "AccessDeniedException";
const _AOA = "ArrayOfArray";
const _AV = "ArrayValue";
const _AVL = "ArrayValueList";
const _BA = "BooleanArray";
const _BES = "BatchExecuteStatement";
const _BESR = "BatchExecuteStatementRequest";
const _BESRa = "BatchExecuteStatementResponse";
const _BRE = "BadRequestException";
const _BT = "BeginTransaction";
const _BTR = "BeginTransactionRequest";
const _BTRe = "BeginTransactionResponse";
const _CM = "ColumnMetadata";
const _CT = "CommitTransaction";
const _CTR = "CommitTransactionRequest";
const _CTRo = "CommitTransactionResponse";
const _DA = "DoubleArray";
const _DEE = "DatabaseErrorException";
const _DNFE = "DatabaseNotFoundException";
const _DRE = "DatabaseResumingException";
const _DUE = "DatabaseUnavailableException";
const _ES = "ExecuteSql";
const _ESR = "ExecuteSqlRequest";
const _ESRx = "ExecuteSqlResponse";
const _ESRxe = "ExecuteStatementRequest";
const _ESRxec = "ExecuteStatementResponse";
const _ESx = "ExecuteStatement";
const _F = "Field";
const _FE = "ForbiddenException";
const _FL = "FieldList";
const _HENEE = "HttpEndpointNotEnabledException";
const _IRSE = "InvalidResourceStateException";
const _ISE = "InvalidSecretException";
const _ISEE = "InternalServerErrorException";
const _LA = "LongArray";
const _M = "Metadata";
const _NFE = "NotFoundException";
const _R = "Record";
const _RF = "ResultFrame";
const _RSM = "ResultSetMetadata";
const _RSO = "ResultSetOptions";
const _RT = "RollbackTransaction";
const _RTR = "RollbackTransactionRequest";
const _RTRo = "RollbackTransactionResponse";
const _Re = "Records";
const _Ro = "Row";
const _SA = "StringArray";
const _SEE = "SecretsErrorException";
const _SP = "SqlParameter";
const _SPL = "SqlParametersList";
const _SPS = "SqlParameterSets";
const _SR = "SqlRecords";
const _SSR = "SqlStatementResult";
const _SSRq = "SqlStatementResults";
const _STE = "StatementTimeoutException";
const _SUE = "ServiceUnavailableError";
const _SV = "StructValue";
const _TNFE = "TransactionNotFoundException";
const _UR = "UpdateResult";
const _URE = "UnsupportedResultException";
const _URp = "UpdateResults";
const _V = "Value";
const _a = "attributes";
const _aBCT = "arrayBaseColumnType";
const _aSSA = "awsSecretStoreArn";
const _aV = "arrayValues";
const _aVr = "arrayValue";
const _bIV = "bigIntValue";
const _bV = "booleanValues";
const _bVi = "bitValue";
const _bVl = "blobValue";
const _bVo = "booleanValue";
const _c = "client";
const _cAT = "continueAfterTimeout";
const _cC = "columnCount";
const _cM = "columnMetadata";
const _d = "database";
const _dCI = "dbConnectionId";
const _dCOIA = "dbClusterOrInstanceArn";
const _dRT = "decimalReturnType";
const _dV = "doubleValues";
const _dVo = "doubleValue";
const _e = "error";
const _fR = "formattedRecords";
const _fRA = "formatRecordsAs";
const _gF = "generatedFields";
const _h = "http";
const _hE = "httpError";
const _iAI = "isAutoIncrement";
const _iC = "isCurrency";
const _iCS = "isCaseSensitive";
const _iN = "isNull";
const _iRM = "includeResultMetadata";
const _iS = "isSigned";
const _iV = "intValue";
const _l = "label";
const _lRT = "longReturnType";
const _lV = "longValues";
const _lVo = "longValue";
const _m = "message";
const _n = "name";
const _nORU = "numberOfRecordsUpdated";
const _nu = "nullable";
const _p = "precision";
const _pS = "parameterSets";
const _pa = "parameters";
const _r = "records";
const _rA = "resourceArn";
const _rF = "resultFrame";
const _rSM = "resultSetMetadata";
const _rSO = "resultSetOptions";
const _rV = "realValue";
const _s = "smithy.ts.sdk.synthetic.com.amazonaws.rdsdata";
const _sA = "secretArn";
const _sN = "schemaName";
const _sS = "sqlStatements";
const _sSR = "sqlStatementResults";
const _sV = "stringValues";
const _sVt = "stringValue";
const _sVtr = "structValue";
const _sc = "schema";
const _sca = "scale";
const _se = "server";
const _sp = "sparse";
const _sq = "sql";
const _t = "type";
const _tH = "typeHint";
const _tI = "transactionId";
const _tN = "typeName";
const _tNa = "tableName";
const _tS = "transactionStatus";
const _uR = "updateResults";
const _v = "values";
const _va = "value";
const n0 = "com.amazonaws.rdsdata";
const _s_registry = TypeRegistry.for(_s);
var RDSDataServiceException$ = [-3, _s, "RDSDataServiceException", 0, [], []];
_s_registry.registerError(RDSDataServiceException$, RDSDataServiceException);
const n0_registry = TypeRegistry.for(n0);
var AccessDeniedException$ = [-3, n0, _ADE,
    { [_e]: _c, [_hE]: 403 },
    [_m],
    [0]
];
n0_registry.registerError(AccessDeniedException$, AccessDeniedException);
var BadRequestException$ = [-3, n0, _BRE,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
n0_registry.registerError(BadRequestException$, BadRequestException);
var DatabaseErrorException$ = [-3, n0, _DEE,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
n0_registry.registerError(DatabaseErrorException$, DatabaseErrorException);
var DatabaseNotFoundException$ = [-3, n0, _DNFE,
    { [_e]: _c, [_hE]: 404 },
    [_m],
    [0]
];
n0_registry.registerError(DatabaseNotFoundException$, DatabaseNotFoundException);
var DatabaseResumingException$ = [-3, n0, _DRE,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
n0_registry.registerError(DatabaseResumingException$, DatabaseResumingException);
var DatabaseUnavailableException$ = [-3, n0, _DUE,
    { [_e]: _se, [_hE]: 504 },
    [],
    []
];
n0_registry.registerError(DatabaseUnavailableException$, DatabaseUnavailableException);
var ForbiddenException$ = [-3, n0, _FE,
    { [_e]: _c, [_hE]: 403 },
    [_m],
    [0]
];
n0_registry.registerError(ForbiddenException$, ForbiddenException);
var HttpEndpointNotEnabledException$ = [-3, n0, _HENEE,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
n0_registry.registerError(HttpEndpointNotEnabledException$, HttpEndpointNotEnabledException);
var InternalServerErrorException$ = [-3, n0, _ISEE,
    { [_e]: _se, [_hE]: 500 },
    [],
    []
];
n0_registry.registerError(InternalServerErrorException$, InternalServerErrorException);
var InvalidResourceStateException$ = [-3, n0, _IRSE,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
n0_registry.registerError(InvalidResourceStateException$, InvalidResourceStateException);
var InvalidSecretException$ = [-3, n0, _ISE,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
n0_registry.registerError(InvalidSecretException$, InvalidSecretException);
var NotFoundException$ = [-3, n0, _NFE,
    { [_e]: _c, [_hE]: 404 },
    [_m],
    [0]
];
n0_registry.registerError(NotFoundException$, NotFoundException);
var SecretsErrorException$ = [-3, n0, _SEE,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
n0_registry.registerError(SecretsErrorException$, SecretsErrorException);
var ServiceUnavailableError$ = [-3, n0, _SUE,
    { [_e]: _se, [_hE]: 503 },
    [],
    []
];
n0_registry.registerError(ServiceUnavailableError$, ServiceUnavailableError);
var StatementTimeoutException$ = [-3, n0, _STE,
    { [_e]: _c, [_hE]: 400 },
    [_m, _dCI],
    [0, 1]
];
n0_registry.registerError(StatementTimeoutException$, StatementTimeoutException);
var TransactionNotFoundException$ = [-3, n0, _TNFE,
    { [_e]: _c, [_hE]: 404 },
    [_m],
    [0]
];
n0_registry.registerError(TransactionNotFoundException$, TransactionNotFoundException);
var UnsupportedResultException$ = [-3, n0, _URE,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
n0_registry.registerError(UnsupportedResultException$, UnsupportedResultException);
const errorTypeRegistries = [
    _s_registry,
    n0_registry,
];
var BatchExecuteStatementRequest$ = [3, n0, _BESR,
    0,
    [_rA, _sA, _sq, _d, _sc, _pS, _tI],
    [0, 0, 0, 0, 0, [() => SqlParameterSets, 0], 0], 3
];
var BatchExecuteStatementResponse$ = [3, n0, _BESRa,
    0,
    [_uR],
    [[() => UpdateResults, 0]]
];
var BeginTransactionRequest$ = [3, n0, _BTR,
    0,
    [_rA, _sA, _d, _sc],
    [0, 0, 0, 0], 2
];
var BeginTransactionResponse$ = [3, n0, _BTRe,
    0,
    [_tI],
    [0]
];
var ColumnMetadata$ = [3, n0, _CM,
    0,
    [_n, _t, _tN, _l, _sN, _tNa, _iAI, _iS, _iC, _iCS, _nu, _p, _sca, _aBCT],
    [0, 1, 0, 0, 0, 0, 2, 2, 2, 2, 1, 1, 1, 1]
];
var CommitTransactionRequest$ = [3, n0, _CTR,
    0,
    [_rA, _sA, _tI],
    [0, 0, 0], 3
];
var CommitTransactionResponse$ = [3, n0, _CTRo,
    0,
    [_tS],
    [0]
];
var ExecuteSqlRequest$ = [3, n0, _ESR,
    0,
    [_dCOIA, _aSSA, _sS, _d, _sc],
    [0, 0, 0, 0, 0], 3
];
var ExecuteSqlResponse$ = [3, n0, _ESRx,
    0,
    [_sSR],
    [() => SqlStatementResults]
];
var ExecuteStatementRequest$ = [3, n0, _ESRxe,
    0,
    [_rA, _sA, _sq, _d, _sc, _pa, _tI, _iRM, _cAT, _rSO, _fRA],
    [0, 0, 0, 0, 0, [() => SqlParametersList, 0], 0, 2, 2, () => ResultSetOptions$, 0], 3
];
var ExecuteStatementResponse$ = [3, n0, _ESRxec,
    0,
    [_r, _cM, _nORU, _gF, _fR],
    [[() => SqlRecords, 0], () => Metadata, 1, [() => FieldList, 0], 0]
];
var _Record$ = [3, n0, _R,
    0,
    [_v],
    [() => Row]
];
var ResultFrame$ = [3, n0, _RF,
    0,
    [_rSM, _r],
    [() => ResultSetMetadata$, () => Records]
];
var ResultSetMetadata$ = [3, n0, _RSM,
    0,
    [_cC, _cM],
    [1, () => Metadata]
];
var ResultSetOptions$ = [3, n0, _RSO,
    0,
    [_dRT, _lRT],
    [0, 0]
];
var RollbackTransactionRequest$ = [3, n0, _RTR,
    0,
    [_rA, _sA, _tI],
    [0, 0, 0], 3
];
var RollbackTransactionResponse$ = [3, n0, _RTRo,
    0,
    [_tS],
    [0]
];
var SqlParameter$ = [3, n0, _SP,
    0,
    [_n, _va, _tH],
    [0, [() => Field$, 0], 0]
];
var SqlStatementResult$ = [3, n0, _SSR,
    0,
    [_rF, _nORU],
    [() => ResultFrame$, 1]
];
var StructValue$ = [3, n0, _SV,
    0,
    [_a],
    [() => ArrayValueList]
];
var UpdateResult$ = [3, n0, _UR,
    0,
    [_gF],
    [[() => FieldList, 0]]
];
var ArrayOfArray = [1, n0, _AOA,
    { [_sp]: 1 }, [() => ArrayValue$,
        0]
];
var ArrayValueList = [1, n0, _AVL,
    0, () => Value$
];
var BooleanArray = [1, n0, _BA,
    { [_sp]: 1 }, 2
];
var DoubleArray = [1, n0, _DA,
    { [_sp]: 1 }, 1
];
var FieldList = [1, n0, _FL,
    0, [() => Field$,
        0]
];
var LongArray = [1, n0, _LA,
    { [_sp]: 1 }, 1
];
var Metadata = [1, n0, _M,
    0, () => ColumnMetadata$
];
var Records = [1, n0, _Re,
    0, () => _Record$
];
var Row = [1, n0, _Ro,
    0, () => Value$
];
var SqlParameterSets = [1, n0, _SPS,
    0, [() => SqlParametersList,
        0]
];
var SqlParametersList = [1, n0, _SPL,
    0, [() => SqlParameter$,
        0]
];
var SqlRecords = [1, n0, _SR,
    0, [() => FieldList,
        0]
];
var SqlStatementResults = [1, n0, _SSRq,
    0, () => SqlStatementResult$
];
var StringArray = [1, n0, _SA,
    { [_sp]: 1 }, 0
];
var UpdateResults = [1, n0, _URp,
    0, [() => UpdateResult$,
        0]
];
var ArrayValue$ = [4, n0, _AV,
    0,
    [_bV, _lV, _dV, _sV, _aV],
    [[() => BooleanArray, 0], [() => LongArray, 0], [() => DoubleArray, 0], [() => StringArray, 0], [() => ArrayOfArray, 0]]
];
var Field$ = [4, n0, _F,
    0,
    [_iN, _bVo, _lVo, _dVo, _sVt, _bVl, _aVr],
    [2, 2, 1, 1, 0, 21, [() => ArrayValue$, 0]]
];
var Value$ = [4, n0, _V,
    0,
    [_iN, _bVi, _bIV, _iV, _dVo, _rV, _sVt, _bVl, _aV, _sVtr],
    [2, 2, 1, 1, 1, 1, 0, 21, () => ArrayValueList, () => StructValue$]
];
var BatchExecuteStatement$ = [9, n0, _BES,
    { [_h]: ["POST", "/BatchExecute", 200] }, () => BatchExecuteStatementRequest$, () => BatchExecuteStatementResponse$
];
var BeginTransaction$ = [9, n0, _BT,
    { [_h]: ["POST", "/BeginTransaction", 200] }, () => BeginTransactionRequest$, () => BeginTransactionResponse$
];
var CommitTransaction$ = [9, n0, _CT,
    { [_h]: ["POST", "/CommitTransaction", 200] }, () => CommitTransactionRequest$, () => CommitTransactionResponse$
];
var ExecuteSql$ = [9, n0, _ES,
    { [_h]: ["POST", "/ExecuteSql", 200] }, () => ExecuteSqlRequest$, () => ExecuteSqlResponse$
];
var ExecuteStatement$ = [9, n0, _ESx,
    { [_h]: ["POST", "/Execute", 200] }, () => ExecuteStatementRequest$, () => ExecuteStatementResponse$
];
var RollbackTransaction$ = [9, n0, _RT,
    { [_h]: ["POST", "/RollbackTransaction", 200] }, () => RollbackTransactionRequest$, () => RollbackTransactionResponse$
];

const getRuntimeConfig$1 = (config) => {
    return {
        apiVersion: "2018-08-01",
        base64Decoder: config?.base64Decoder ?? fromBase64,
        base64Encoder: config?.base64Encoder ?? toBase64,
        disableHostPrefix: config?.disableHostPrefix ?? false,
        endpointProvider: config?.endpointProvider ?? defaultEndpointResolver,
        extensions: config?.extensions ?? [],
        httpAuthSchemeProvider: config?.httpAuthSchemeProvider ?? defaultRDSDataHttpAuthSchemeProvider,
        httpAuthSchemes: config?.httpAuthSchemes ?? [
            {
                schemeId: "aws.auth#sigv4",
                identityProvider: (ipc) => ipc.getIdentityProvider("aws.auth#sigv4"),
                signer: new AwsSdkSigV4Signer(),
            },
        ],
        logger: config?.logger ?? new NoOpLogger(),
        protocol: config?.protocol ?? AwsRestJsonProtocol,
        protocolSettings: config?.protocolSettings ?? {
            defaultNamespace: "com.amazonaws.rdsdata",
            errorTypeRegistries,
            version: "2018-08-01",
            serviceTarget: "RdsDataService",
        },
        serviceId: config?.serviceId ?? "RDS Data",
        sha256: config?.sha256 ?? Sha256,
        urlParser: config?.urlParser ?? parseUrl,
        utf8Decoder: config?.utf8Decoder ?? fromUtf8,
        utf8Encoder: config?.utf8Encoder ?? toUtf8,
    };
};

const getRuntimeConfig = (config) => {
    emitWarningIfUnsupportedVersion(process.version);
    const defaultsMode = resolveDefaultsModeConfig(config);
    const defaultConfigProvider = () => defaultsMode().then(loadConfigsForDefaultMode);
    const clientSharedValues = getRuntimeConfig$1(config);
    emitWarningIfUnsupportedVersion$1(process.version);
    const loaderConfig = {
        profile: config?.profile,
        logger: clientSharedValues.logger,
    };
    return {
        ...clientSharedValues,
        ...config,
        runtime: "node",
        defaultsMode,
        authSchemePreference: config?.authSchemePreference ?? loadConfig(NODE_AUTH_SCHEME_PREFERENCE_OPTIONS, loaderConfig),
        bodyLengthChecker: config?.bodyLengthChecker ?? calculateBodyLength,
        credentialDefaultProvider: config?.credentialDefaultProvider ?? defaultProvider,
        defaultUserAgentProvider: config?.defaultUserAgentProvider ?? createDefaultUserAgentProvider({ serviceId: clientSharedValues.serviceId, clientVersion: packageInfo.version }),
        maxAttempts: config?.maxAttempts ?? loadConfig(NODE_MAX_ATTEMPT_CONFIG_OPTIONS, config),
        region: config?.region ?? loadConfig(NODE_REGION_CONFIG_OPTIONS, { ...NODE_REGION_CONFIG_FILE_OPTIONS, ...loaderConfig }),
        requestHandler: NodeHttpHandler.create(config?.requestHandler ?? defaultConfigProvider),
        retryMode: config?.retryMode ??
            loadConfig({
                ...NODE_RETRY_MODE_CONFIG_OPTIONS,
                default: async () => (await defaultConfigProvider()).retryMode || DEFAULT_RETRY_MODE,
            }, config),
        streamCollector: config?.streamCollector ?? streamCollector,
        useDualstackEndpoint: config?.useDualstackEndpoint ?? loadConfig(NODE_USE_DUALSTACK_ENDPOINT_CONFIG_OPTIONS, loaderConfig),
        useFipsEndpoint: config?.useFipsEndpoint ?? loadConfig(NODE_USE_FIPS_ENDPOINT_CONFIG_OPTIONS, loaderConfig),
        userAgentAppId: config?.userAgentAppId ?? loadConfig(NODE_APP_ID_CONFIG_OPTIONS, loaderConfig),
    };
};

const getHttpAuthExtensionConfiguration = (runtimeConfig) => {
    const _httpAuthSchemes = runtimeConfig.httpAuthSchemes;
    let _httpAuthSchemeProvider = runtimeConfig.httpAuthSchemeProvider;
    let _credentials = runtimeConfig.credentials;
    return {
        setHttpAuthScheme(httpAuthScheme) {
            const index = _httpAuthSchemes.findIndex((scheme) => scheme.schemeId === httpAuthScheme.schemeId);
            if (index === -1) {
                _httpAuthSchemes.push(httpAuthScheme);
            }
            else {
                _httpAuthSchemes.splice(index, 1, httpAuthScheme);
            }
        },
        httpAuthSchemes() {
            return _httpAuthSchemes;
        },
        setHttpAuthSchemeProvider(httpAuthSchemeProvider) {
            _httpAuthSchemeProvider = httpAuthSchemeProvider;
        },
        httpAuthSchemeProvider() {
            return _httpAuthSchemeProvider;
        },
        setCredentials(credentials) {
            _credentials = credentials;
        },
        credentials() {
            return _credentials;
        },
    };
};
const resolveHttpAuthRuntimeConfig = (config) => {
    return {
        httpAuthSchemes: config.httpAuthSchemes(),
        httpAuthSchemeProvider: config.httpAuthSchemeProvider(),
        credentials: config.credentials(),
    };
};

const resolveRuntimeExtensions = (runtimeConfig, extensions) => {
    const extensionConfiguration = Object.assign(getAwsRegionExtensionConfiguration(runtimeConfig), getDefaultExtensionConfiguration(runtimeConfig), getHttpHandlerExtensionConfiguration(runtimeConfig), getHttpAuthExtensionConfiguration(runtimeConfig));
    extensions.forEach((extension) => extension.configure(extensionConfiguration));
    return Object.assign(runtimeConfig, resolveAwsRegionExtensionConfiguration(extensionConfiguration), resolveDefaultRuntimeConfig(extensionConfiguration), resolveHttpHandlerRuntimeConfig(extensionConfiguration), resolveHttpAuthRuntimeConfig(extensionConfiguration));
};

class RDSDataClient extends Client {
    config;
    constructor(...[configuration]) {
        const _config_0 = getRuntimeConfig(configuration || {});
        super(_config_0);
        this.initConfig = _config_0;
        const _config_1 = resolveClientEndpointParameters(_config_0);
        const _config_2 = resolveUserAgentConfig(_config_1);
        const _config_3 = resolveRetryConfig(_config_2);
        const _config_4 = resolveRegionConfig(_config_3);
        const _config_5 = resolveHostHeaderConfig(_config_4);
        const _config_6 = resolveEndpointConfig(_config_5);
        const _config_7 = resolveHttpAuthSchemeConfig(_config_6);
        const _config_8 = resolveRuntimeExtensions(_config_7, configuration?.extensions || []);
        this.config = _config_8;
        this.middlewareStack.use(getSchemaSerdePlugin(this.config));
        this.middlewareStack.use(getUserAgentPlugin(this.config));
        this.middlewareStack.use(getRetryPlugin(this.config));
        this.middlewareStack.use(getContentLengthPlugin(this.config));
        this.middlewareStack.use(getHostHeaderPlugin(this.config));
        this.middlewareStack.use(getLoggerPlugin(this.config));
        this.middlewareStack.use(getRecursionDetectionPlugin(this.config));
        this.middlewareStack.use(getHttpAuthSchemeEndpointRuleSetPlugin(this.config, {
            httpAuthSchemeParametersProvider: defaultRDSDataHttpAuthSchemeParametersProvider,
            identityProviderConfigProvider: async (config) => new DefaultIdentityProviderConfig({
                "aws.auth#sigv4": config.credentials,
            }),
        }));
        this.middlewareStack.use(getHttpSigningPlugin(this.config));
    }
    destroy() {
        super.destroy();
    }
}

const command = makeBuilder(commonParams, "RdsDataService", "RDSDataClient", getEndpointPlugin);
const _ep0 = {};
const _mw0 = (Command, cs, config, o) => [];

class BatchExecuteStatementCommand extends command(_ep0, _mw0, "BatchExecuteStatement", BatchExecuteStatement$) {
}

class BeginTransactionCommand extends command(_ep0, _mw0, "BeginTransaction", BeginTransaction$) {
}

class CommitTransactionCommand extends command(_ep0, _mw0, "CommitTransaction", CommitTransaction$) {
}

class ExecuteSqlCommand extends command(_ep0, _mw0, "ExecuteSql", ExecuteSql$) {
}

class ExecuteStatementCommand extends command(_ep0, _mw0, "ExecuteStatement", ExecuteStatement$) {
}

class RollbackTransactionCommand extends command(_ep0, _mw0, "RollbackTransaction", RollbackTransaction$) {
}

const commands = {
    BatchExecuteStatementCommand,
    BeginTransactionCommand,
    CommitTransactionCommand,
    ExecuteSqlCommand,
    ExecuteStatementCommand,
    RollbackTransactionCommand,
};
class RDSData extends RDSDataClient {
}
createAggregatedClient(commands, RDSData);

const TypeHint = {
    DATE: "DATE",
    DECIMAL: "DECIMAL",
    JSON: "JSON",
    TIME: "TIME",
    TIMESTAMP: "TIMESTAMP",
    UUID: "UUID",
};
const DecimalReturnType = {
    DOUBLE_OR_LONG: "DOUBLE_OR_LONG",
    STRING: "STRING",
};
const RecordsFormatType = {
    JSON: "JSON",
    NONE: "NONE",
};
const LongReturnType = {
    LONG: "LONG",
    STRING: "STRING",
};

exports.AccessDeniedException = AccessDeniedException;
exports.AccessDeniedException$ = AccessDeniedException$;
exports.ArrayValue$ = ArrayValue$;
exports.BadRequestException = BadRequestException;
exports.BadRequestException$ = BadRequestException$;
exports.BatchExecuteStatement$ = BatchExecuteStatement$;
exports.BatchExecuteStatementCommand = BatchExecuteStatementCommand;
exports.BatchExecuteStatementRequest$ = BatchExecuteStatementRequest$;
exports.BatchExecuteStatementResponse$ = BatchExecuteStatementResponse$;
exports.BeginTransaction$ = BeginTransaction$;
exports.BeginTransactionCommand = BeginTransactionCommand;
exports.BeginTransactionRequest$ = BeginTransactionRequest$;
exports.BeginTransactionResponse$ = BeginTransactionResponse$;
exports.ColumnMetadata$ = ColumnMetadata$;
exports.CommitTransaction$ = CommitTransaction$;
exports.CommitTransactionCommand = CommitTransactionCommand;
exports.CommitTransactionRequest$ = CommitTransactionRequest$;
exports.CommitTransactionResponse$ = CommitTransactionResponse$;
exports.DatabaseErrorException = DatabaseErrorException;
exports.DatabaseErrorException$ = DatabaseErrorException$;
exports.DatabaseNotFoundException = DatabaseNotFoundException;
exports.DatabaseNotFoundException$ = DatabaseNotFoundException$;
exports.DatabaseResumingException = DatabaseResumingException;
exports.DatabaseResumingException$ = DatabaseResumingException$;
exports.DatabaseUnavailableException = DatabaseUnavailableException;
exports.DatabaseUnavailableException$ = DatabaseUnavailableException$;
exports.DecimalReturnType = DecimalReturnType;
exports.ExecuteSql$ = ExecuteSql$;
exports.ExecuteSqlCommand = ExecuteSqlCommand;
exports.ExecuteSqlRequest$ = ExecuteSqlRequest$;
exports.ExecuteSqlResponse$ = ExecuteSqlResponse$;
exports.ExecuteStatement$ = ExecuteStatement$;
exports.ExecuteStatementCommand = ExecuteStatementCommand;
exports.ExecuteStatementRequest$ = ExecuteStatementRequest$;
exports.ExecuteStatementResponse$ = ExecuteStatementResponse$;
exports.Field$ = Field$;
exports.ForbiddenException = ForbiddenException;
exports.ForbiddenException$ = ForbiddenException$;
exports.HttpEndpointNotEnabledException = HttpEndpointNotEnabledException;
exports.HttpEndpointNotEnabledException$ = HttpEndpointNotEnabledException$;
exports.InternalServerErrorException = InternalServerErrorException;
exports.InternalServerErrorException$ = InternalServerErrorException$;
exports.InvalidResourceStateException = InvalidResourceStateException;
exports.InvalidResourceStateException$ = InvalidResourceStateException$;
exports.InvalidSecretException = InvalidSecretException;
exports.InvalidSecretException$ = InvalidSecretException$;
exports.LongReturnType = LongReturnType;
exports.NotFoundException = NotFoundException;
exports.NotFoundException$ = NotFoundException$;
exports.RDSData = RDSData;
exports.RDSDataClient = RDSDataClient;
exports.RDSDataServiceException = RDSDataServiceException;
exports.RDSDataServiceException$ = RDSDataServiceException$;
exports.RecordsFormatType = RecordsFormatType;
exports.ResultFrame$ = ResultFrame$;
exports.ResultSetMetadata$ = ResultSetMetadata$;
exports.ResultSetOptions$ = ResultSetOptions$;
exports.RollbackTransaction$ = RollbackTransaction$;
exports.RollbackTransactionCommand = RollbackTransactionCommand;
exports.RollbackTransactionRequest$ = RollbackTransactionRequest$;
exports.RollbackTransactionResponse$ = RollbackTransactionResponse$;
exports.SecretsErrorException = SecretsErrorException;
exports.SecretsErrorException$ = SecretsErrorException$;
exports.ServiceUnavailableError = ServiceUnavailableError;
exports.ServiceUnavailableError$ = ServiceUnavailableError$;
exports.SqlParameter$ = SqlParameter$;
exports.SqlStatementResult$ = SqlStatementResult$;
exports.StatementTimeoutException = StatementTimeoutException;
exports.StatementTimeoutException$ = StatementTimeoutException$;
exports.StructValue$ = StructValue$;
exports.TransactionNotFoundException = TransactionNotFoundException;
exports.TransactionNotFoundException$ = TransactionNotFoundException$;
exports.TypeHint = TypeHint;
exports.UnsupportedResultException = UnsupportedResultException;
exports.UnsupportedResultException$ = UnsupportedResultException$;
exports.UpdateResult$ = UpdateResult$;
exports.Value$ = Value$;
exports._Record$ = _Record$;
exports.errorTypeRegistries = errorTypeRegistries;
